### PR TITLE
Prevent JobSpec params from overwriting TaskSpec params

### PR DESCRIPTION
**Pull request by j16r**
Signed-off-by: John Barker <jebarker@gmail.com>

### CODE REVIEW

File: services/job_runner_test.go [hunk 1]
```diff
@@ -149,13 +149,14 @@ func TestJobRunner_ExecuteRun(t *testing.T) {
...
 		wantData   string
+		wantBody   string
 	}{
-		{"success", bridgeName, `{}`, `{"data":{"value":"100"}}`, models.RunStatusCompleted, `{"value":"100","type":"auctionBidding"}`},
-		{"errored", bridgeName, `{}`, `{"error":"too much"}`, models.RunStatusErrored, `{"type":"auctionBidding"}`},
-		{"errored with a value", bridgeName, `{}`, `{"error":"too much", "data":{"value":"99"}}`, models.RunStatusErrored, `{"value":"99","type":"auctionBidding"}`},
-		{"overriding bridge type params", bridgeName, `{"url":"hack"}`, `{"data":{"value":"100"}}`, models.RunStatusCompleted, `{"value":"100","url":"hack","type":"auctionBidding"}`},
-		{"type parameter does not override", bridgeName, `{"type":"0"}`, `{"data":{"value":"100"}}`, models.RunStatusCompleted, `{"value":"100","type":"0"}`},
-		{"non-existent bridge type", "non-existent", `{}`, `{}`, models.RunStatusErrored, `{}`},
+		{"success", bridgeName, `{}`, `{"data":{"value":"100"}}`, models.RunStatusCompleted, `{"value":"100","type":"auctionBidding"}`, `{"type":"auctionBidding"}`},
```
Comment (j16r) [inline services/job_runner_test.go]: @se3000/@thodges-gh  should wantBody contain `"value":"100"` ?

Comment (se3000) [reply services/job_runner_test.go]: Yes, and `"type":"auctionBidding"`.

Comment (j16r) [reply services/job_runner_test.go]: Cool, looks like the tests weren't asserting for this and it was confusing us. Will fix in this PR.

Comment (j16r) [reply services/job_runner_test.go]: Hmm, circling back on this - `"value":"100"` is contained in the response body of the bridgeAdapter, so it cannot be submitted to the bridgeAdapter (without violating causality).

I believe these tests are all fine, I've just changed them to be more explicit.

**Comment by thodges-gh** [comment]
So I found some disparate behavior between how the task params are merged.

At 78253e6, when I start a task run with a web initiator, everything works as expected. Given a job spec like this:
```json
{
  "initiators": [
    { "type": "web" }
  ],
  "tasks": [
    { "type": "ea", "my_param": "some_value" }
  ]
}
```

If I try to override the `"my_param"` value in a run, the JobSpec parameter is given preference, as we prefer. Input:

```json
{"my_param":"overridden_value", "new_param":"new_value"}
```

Adapter sees:
```json
{"id":"e0d7abd6c83f43b2a92746073192e246","data":{"confirmations":0,"my_param":"some_value","new_param":"new_value","type":"ea"}}
```

**However,** when a JobSpec uses the RunLog initiator, the given task parameters do still override the parameters set in the JobSpec:

My JobSpec (created with params specified):
```json
{
  "initiators": [
    {
      "type": "runlog",
      "address": "0xf41adafc551037ba549a38698c38c88d24eb10e6"
    }
  ],
  "tasks": [
    {
      "confirmations": 0,
      "type": "httpget",
      "url": "https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD"
    },
    {
      "confirmations": 0,
      "path": [
        "USD"
      ],
      "type": "jsonparse"
    },
    {
      "confirmations": 0,
      "times": 10000,
      "type": "multiply"
    },
    {
      "confirmations": 0,
      "type": "ethuint256"
    },
    {
      "confirmations": 0,
      "type": "ethtx"
    }
  ],
  "startAt": null,
  "endAt": null
}
```

The request method:

```
  function requestEthereumPrice() 
    public
    onlyOwner
  {
    ChainlinkLib.Run memory run = newRun(SPEC_ID, this, "fulfillEthereumPrice(bytes32,uint256)");
    run.add("url", "https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD,EUR,JPY");
    string[] memory path = new string[](1);
    path[0] = "USD";
    run.addStringArray("path", path);
    run.add("someExtra", "my_value");
    run.addInt("times", 100);
    chainlinkRequest(run, LINK(1));
  }
```

Task Run Params (notice the url and times values are overridden, but "someExtra" is still present, which is good):
```json
{
  "address": "0xf41adAfc551037bA549a38698c38c88D24eB10e6",
  "confirmations": 0,
  "dataPrefix": "0x00568ea2c86678a0bececfd9086b2d636ebcf854587bf387727a0770e455e66e",
  "functionSelector": "0x76005c26",
  "path": [
    "USD"
  ],
  "someExtra": "my_value",
  "times": 100,
  "type": "ethtx",
  "url": "https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD,EUR,JPY",
  "value": "0xd45d9a34a6498436477533e7080779c0f5f9e73685eba4151df53f5e9be8f86f"
}
```

What should happen here is any field which would override an existing parameter does not, and any additional field (which did not exist in the original spec) is added, just like it currently works with a web initiator request.

**Comment by j16r** [comment]
@thodges-gh since this fixes your specific problem, shall we merge and I'll look into exploring the logging messages in a separate PR?

**Pull request merged**